_posts/2024-10-10-jakarta-data.adoc

```diff
@@ -0,0 +1,227 @@
+
+The repository interface methods are annotated with various Jakarta Data annotations (`@Insert`, `@Find`, etc) that define the expected persistence behavior of the method.
+
+There's much more to the Jakarta Data programming model than this; for all the details see:
+
+* The link:https://jakarta.ee/specifications/data/1.0/jakarta-data-1.0[Jakarta Data 1.0 specification, window=_blank]
+* The link:https://docs.jboss.org/hibernate/orm/6.6/repositories/html_single/Hibernate_Data_Repositories.html[Hibernate Data Repositories documentation, window=_blank]
+* Gavin King's excellent link:https://in.relation.to/2024/04/01/jakarta-data-1/[blog posts, window=_blank] on Jakarta Data
+
+
+A Jakarta Data implementation like WildFly Preview can support one or more Jakarta Data link:https://jakarta.ee/specifications/data/1.0/jakarta-data-1.0#_jakarta_data_providers[providers, window=_blank]. A provider understands one or more Java annotation types that are used to define entities, and it understands how to interact with a particular type of back end datastore.
+
+WildFly Preview's Jakarta Data implementation supports the link:https://docs.jboss.org/hibernate/orm/6.6/repositories/html_single/Hibernate_Data_Repositories.html[Hibernate Data Repositories, window=_blank] provider, which uses Hibernate ORM to interact with a variety of different relational databases. Hibernate Data Repositories supports the `jakarta.persistence.Entity` annotation as the mechanism for application authors to define entities.
+
+== Using Hibernate Data Repositories in Your Application
+
+There are two key things to understand in order to use WildFly Preview's Hibernate Data Repositories provider:
+
+* How to configure build time generation of the implementation of your `@Repository` interfaces.
+* How to configure the runtime behavior of the Hibernate ORM instance that will interact with the database.
+
+
+=== Build-time Generation of Repository Implementations
+
+An application author using Jakarta Data simply writes an interface for their repository, but of course for that to work at runtime there must be an actual implementation of that interface. It's the responsibility of the Jakarta Data provider to provide that implementation. Hibernate Data Repositories does this by generating the implementation classes as part of the build of your application.
+
+So, to use Jakarta Data with WildFly Preview you need to configure the generation of those classes as part of your application build. In a Maven build this is done configuring the Maven compiler plugin to use the `org.hibernate.orm:hibernate-jpamodelgen` artifact as an annotation processor:
+
+[source,xml]
+----
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.12.0</version> <!-- 3.12 or later is necessary for proper dependency management in annotationProcessorPaths -->
```

Comment:
Might as well recommend the latest 3.12 while you're at it; 3.12.0 had some rather annoying issues that did impact other projects: https://issues.apache.org/jira/browse/MCOMPILER-567

```suggestion
                <version>3.12.1</version> <!-- 3.12 or later is necessary for proper dependency management in annotationProcessorPaths -->
```